### ISSUE TITLE
Object detection

### ISSUE THREAD

**Issue by 4F2E4A2E** [bodokaiser/piwise#12]
Hi there and thank you for piwise!

My question is:  how could piwise or any segnet implementation be used for object detection?

**Comment by bodokaiser** [comment]
Depends on the object(s)?

You could train on your own dataset with custom classes wherein every class corresponds to the object you want to detect.

This however require you to have the labeled data.

**Comment by 4F2E4A2E** [comment]
Can you please explain on how it would depend on the object?

**Comment by bodokaiser** [comment]
From what should the network know if you want to detect an apple or a car?

**Comment by 4F2E4A2E** [comment]
Please elaborate, could any network know any difference besides the information given by the semantic segmentation labeling?

**Comment by bodokaiser** [comment]
You can’t say for sure what happens in the hidden layers but generally the network will only output what it was trained with.

**Comment by 4F2E4A2E** [comment]
Exactly, that's my honest opinion too, but my question is, how to train _with_ the segmented ground truth data?

**Comment by bodokaiser** [comment]
You should checkout „fine-tuning“.

**Comment by 4F2E4A2E** [comment]
This one? https://arxiv.org/pdf/1601.05150.pdf

**Comment by bodokaiser** [comment]
Yes that is one application. Actually „fine-tuning“ is quite a common term in deep learning:

http://pytorch.org/tutorials/beginner/transfer_learning_tutorial.html

It is also required by some of the semantic segmentation networks which use a vgg trained network and fine tune it to the PASCAL VOC class labels.
